src/Tribe/Shortcodes/User_Event_Confirmation_List.php

```diff
@@ -80,102 +91,87 @@ protected function generate_attendance_list() {
-			  ON event_end_dates.post_id = ID
-
-			WHERE (
-				-- Match the user
-				match_user.meta_key = '_tribe_tickets_attendee_user_id'
-				AND match_user.meta_value = %d
-			) AND (
-				-- Restrict to upcoming events
-				match_events.meta_key IN ( $keys )
-				AND event_end_dates.meta_key = '_EventEndDateUTC'
-				AND event_end_dates.meta_value > %s
-			)
-			AND NOT EXISTS (
-				SELECT 1
-				  FROM {$wpdb->postmeta} ticket_status
-				 WHERE ticket_status.meta_key = '_wp_trash_meta_status'
-				   AND ticket_status.post_id = match_events.post_id
-			)
-			GROUP BY ID, event_end_dates.meta_value
-			ORDER BY event_end_dates.meta_value
-
-			$limit
-		";
-
-		return (array) $wpdb->get_col( $wpdb->prepare(
-			$query,
-			absint( $this->params['user'] ),
-			current_time( 'mysql' )
-		) );
-	}
+		if ( empty( $this->params['user'] ) ) {
+			return [];
+		}
 
-	/**
-	 * Provides an array containing the value of the ATTENDEE_EVENT_KEY class constant
-	 * for each active ticketing provider.
-	 *
-	 * @return array
-	 */
-	protected function get_event_keys() {
-		$event_keys = array();
-
-		foreach ( Tribe__Tickets__Tickets::modules() as $module_class => $module_instance ) {
-			/**
-			 * The usage of plain `$module_class::ATTENDEE_EVENT_KEY` will throw a `T_PAAMAYIM_NEKUDOTAYIM`
-			 * when using PHP 5.2, which is a fatal.
-			 *
-			 * So we have to construct the constant name using a string and use the `constant` function.
-			 */
-			if ( defined( "$module_class::ATTENDEE_EVENT_KEY" ) ) {
-				$event_keys[] = constant( "$module_class::ATTENDEE_EVENT_KEY" );
-			} else {
-				$event_keys[] = call_user_func( array( $module_class, 'get_key' ), 'ATTENDEE_EVENT_KEY' );
+		$post_ids                    = [];
+		$ticket_ids                  = [];
+		$event_ids                   = [];
+		$upcoming_non_event_post_ids = [];
+
+		// Get all of this user's attendee records
+
+		/** @var Tribe__Tickets__Attendee_Repository $attendees */
+		$attendees = tribe_attendees();
+
+		$attendee_ids = $attendees
+			->by( 'user', $this->params['user'] )
+			->fields( 'ids' )
+			->get_ids();
+
+		// Build list of Posts and Tickets this user's attendance applies to
+
+		/** @var Tribe__Tickets__Tickets_Handler $tickets_handler */
+		$tickets_handler = tribe( 'tickets.handler' );
+
+		foreach ( $attendee_ids as $attendee_id ) {
+			$connection = $tickets_handler->get_object_connections( $attendee_id );
+
+			if ( ! empty( $connection->event ) ) {
+				$post_ids[] = $connection->event;
+			}
+
+			if ( ! empty( $connection->product ) ) {
+				$ticket_ids[] = $connection->product;
 			}
 		}
 
-		return $event_keys;
-	}
+		$post_ids = array_unique( $post_ids );
 
-	/**
-	 * Provides a quoted, comma separated and escaped list of meta keys used to link
-	 * attendee posts to event posts.
-	 *
-	 * @return string
-	 */
-	protected function build_escaped_key_list( array $keys ) {
-		global $wpdb;
+		// Restrict to upcoming Tribe Events posts
```

Comment:
:no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: 6c7ae81b95ca276cf561430a98c4afae</sub>